nexus/pkg/server/sender.go

```diff
@@ -136,6 +136,7 @@ func NewSender(ctx context.Context, settings *service.Settings, logger *observab
 		)
 		uploaderRetryClient := clients.NewRetryClient(
 			clients.WithRetryClientLogger(logger),
+			clients.WithRetryClientRetryPolicy(clients.CheckRetry),
```

Comment:
please review the python retry logic for the file_pusher and make sure that the logic you add here matches the one in the python implementation. 
I pretty sure that the GQL client have different policy than the file uploader.